File: 169. Majority Element/solution.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        Dictionary<int,int> items = new Dictionary<int,int>();
        int maxKey = -1;
        int maxValue = 0;
        
        foreach(int x in nums){
            if(items.ContainsKey(x))
                items[x] += 1;
            else
                items.Add(x,1);
            
            if(items[x] > maxValue){
                maxValue = items[x];
                maxKey = x;
            }
        }
        
        
        return maxKey;
    }
}

File: 169. Majority Element/solution.php
<?php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function majorityElement($nums) {
        $res = [];
        $more_repeat_key = null;
        $more_repeat_count = 0;
        foreach($nums as $n){
            if(isset($res[$n]))
                $res[$n] += 1;
            else
                $res[$n] = 1;

            if($res[$n] > $more_repeat_count)                {
                $more_repeat_count = $res[$n];
                $more_repeat_key = $n;
            }
        }

        return $more_repeat_key;
    }
}
